core/modules/replica/AdminApp.cc

```diff
@@ -0,0 +1,296 @@
+/*
+ * LSST Data Management System
+ * Copyright 2018 LSST Corporation.
+ *
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <http://www.lsstcorp.org/LegalNotices/>.
+ */
+
+// Class header
+#include "replica/AdminApp.h"
+
+// System headers
+#include <iomanip>
+#include <iostream>
+#include <stdexcept>
+#include <vector>
+
+// Qserv headers
+#include "replica/Controller.h"
+#include "replica/Performance.h"
+#include "replica/RequestTracker.h"
+#include "replica/ServiceManagementRequest.h"
+#include "util/TablePrinter.h"
+
+using namespace std;
+
+namespace {
+
+string const description {
+    "This is a Controller application for launching worker management requests."
+};
+
+} /// namespace
+
+
+namespace lsst {
+namespace qserv {
+namespace replica {
+
+AdminApp::Ptr AdminApp::create(int argc,
+                               const char* const argv[]) {
+    return Ptr(
+        new AdminApp(
+            argc,
+            argv
+        )
+    );
+}
+
+
+AdminApp::AdminApp(int argc,
+                   const char* const argv[])
+    :   Application(
+            argc,
+            argv,
+            ::description,
+            true    /* injectDatabaseOptions */,
+            true    /* boostProtobufVersionCheck */,
+            true    /* enableServiceProvider */
```

Comment:
I was just a bit lazy. I could as well do the following in C++ which would look similar to Python:
```
    :   Application(
            argc, argv,
            ::description,
            injectDatabaseOptions(true),
            boostProtobufVersionCheck(true),
            enableServiceProvider(true))
```
This will just require writing 3 functions returning objects of the corresponding classes (structures) for type safety:
```
struct InjectDatabaseOptions {
    bool value = false;
};
InjectDatabaseOptions  injectDatabaseOptions(bool value) {
    return InjectDatabaseOptions{value};
}
```
And then have the constructor of class Application to accept the following parameters:
```
   Application(
      ...
      InjectDatabaseOptions const& injectDatabaseOptions_,
      ...
   );
```

Reply:
I can do tis if you agree that what I'm proposing above makes a sense.

Reply:
Ulike Python, the parameters will still be positional. Though, with a little bit more work I can see ways to make them positionally independent too.

Reply:
It is really up to you, I was trying something like that in my other project but then decided that it only adds to noise. Right answer is of course :snake: :wink: